src/components/application_manager/test/commands/mobile/alert_request_test.cc

```diff
@@ -204,32 +251,475 @@ TEST_F(AlertRequestTest, Run_MandatoryParametersAreInvalid_UNSUCCESS) {
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(mock_app_manager_, ManageHMICommand(_))
+      .Times(AtLeast(1))
+      .WillRepeatedly(Return(true));
+
+  command->Run();
+
+  Event ui_event(hmi_apis::FunctionID::UI_Alert);
+  ui_event.set_smart_object(*msg_);
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageHMICommand(HMIResultCodeIs(hmi_apis::FunctionID::TTS_StopSpeaking)))
+      .WillOnce(Return(true));
+
+  command->on_event(ui_event);
+
+  Event tts_stop_event(hmi_apis::FunctionID::TTS_StopSpeaking);
+  tts_stop_event.set_smart_object(*msg_);
+  command->on_event(tts_stop_event);
+
+  Event event(hmi_apis::FunctionID::TTS_Speak);
+  event.set_smart_object(*msg_);
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_apis::Result::WARNINGS),
+                          am::commands::Command::CommandOrigin::ORIGIN_SDL));
+
+  command->on_event(event);
+}
+
+TEST_F(AlertRequestTest, Run_InvalidAlert2_UNSUCCESS) {
+  AddAlertTextsToMsg();
+  (*msg_)[am::strings::msg_params][am::strings::alert_text2] =
+      "invalid_text_with_empty_str\\n";
+
+  CommandPtr command(CreateCommand<AlertRequest>(msg_));
+
+  ON_CALL(*mock_app_, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_BACKGROUND));
+
+  EXPECT_CALL(mock_message_helper_, ProcessSoftButtons(_, _, _, _)).Times(0);
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_apis::Result::INVALID_DATA),
+                          am::commands::Command::CommandOrigin::ORIGIN_SDL));
+
+  command->Run();
+}
+
+TEST_F(AlertRequestTest, Run_InvalidAlert3_UNSUCCESS) {
+  AddAlertTextsToMsg();
+  (*msg_)[am::strings::msg_params][am::strings::alert_text3] =
+      "invalid_text_with_empty_str\\n";
+
+  CommandPtr command(CreateCommand<AlertRequest>(msg_));
+
+  ON_CALL(*mock_app_, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_BACKGROUND));
+
+  EXPECT_CALL(mock_message_helper_, ProcessSoftButtons(_, _, _, _)).Times(0);
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_apis::Result::INVALID_DATA),
+                          am::commands::Command::CommandOrigin::ORIGIN_SDL));
+
+  command->Run();
+}
+
+TEST_F(AlertRequestTest, Run_InvalidTTSChunk_UNSUCCESS) {
+  AddAlertTextsToMsg();
+  (*msg_)[am::strings::msg_params][am::strings::tts_chunks][0]
+         [am::strings::text] = "invalid_text_with_empty_str\\n";
+
+  CommandPtr command(CreateCommand<AlertRequest>(msg_));
+
+  ON_CALL(*mock_app_, hmi_level())
+      .WillByDefault(Return(mobile_apis::HMILevel::HMI_BACKGROUND));
+
+  EXPECT_CALL(mock_message_helper_, ProcessSoftButtons(_, _, _, _)).Times(0);
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_apis::Result::INVALID_DATA),
+                          am::commands::Command::CommandOrigin::ORIGIN_SDL));
+
+  command->Run();
 }
```

Comment:
@okozlovlux Again code repetition, only thing that is different in the last 3 tests is the message parameter that you are setting.